### ISSUE TITLE
Shavit-misc breaking triggers

### ISSUE THREAD

**Issue by m-yk** [shavitush/bhoptimer#523]
Gotten a few complaints about this from players and decided to do some testing. Went through the AM thread to find someone else was having the issue (my post [here](https://forums.alliedmods.net/showpost.php?p=2559897&postcount=1129)) & tried sneaK's suggestion of removing MPBH, problem still occurred.

saw some people mentioning things about shavit-misc so i tested unloading/loading that both before & after map load and got some inconsistent results but can confirm that if shavit-misc is loaded with the mapchange the triggers break somewhat "consistently". 

i have a really shitty video which should go into a bit more detail https://www.youtube.com/watch?v=uyARRtax7NQ

discord: myk#4418

**Comment by shavitush** [comment]
Thanks! I'll investigate ASAP.

**Comment by strafe** [comment]
Just going to note I've seen this on many servers and multiple timers. I specifically remember it appears in [this video](https://youtu.be/y3IQJAn5nG4?t=1m41s) a few times and this is from SourceCode a couple months before it closed.

The invisible blocks seem to happen when spectating a bot specifically, I've never seen the non-working trigger though.

**Comment by shavitush** [comment]
Alright, so this seems to be an issue with `targetname` resets. Setting `shavit_misc_resettargetname` to 0 fixes it. I'll push a commit that changes the default value to 0 and an updated description. The change will not roll for old setups, so you'll have to change it in your configuration too.

Technical info: some mappers use filter based bhop triggers. Upon testing in bhop_downtown, it seems that the very first platform of the map uses a trigger that changes your targetname to `fil_fw` upon touching the trigger for the first time rather than continuously update the targetname.

![image](https://user-images.githubusercontent.com/3672466/32811071-7f701204-c9a6-11e7-9232-4acad8762b60.png)

And because of the trigger setup of the map, some `trigger_teleport`s require your targetname to be something specific to work.

TL;DR leave it disabled and only enable it for specific cases (such as bhop_strafe_fix to avoid this kind of skips: https://youtu.be/FvGbGyzWhUY?t=440)

**Comment by m-yk** [comment]
`shavit_misc_resettargetname 0` has fixed the issue on downtown & all the other maps I knew of. 👍 
will look at setting configs for maps as I get a list of them

**Comment by Lime18** [comment]
The problem still remains (timer last version)...
"shavit_misc_resettargetname" = "0" min. 0.000000 max. 1.000000 - Reset the player's targetname on timer start?
Recommended to leave disabled. E


On the old version of the timer where there was no support for the late loading, **sm plugins reload reloaded shavit-zones.smx** (gamemode_casual_server.cfg) solved the problem.

In the latest version of the timer it does not help ...

**Comment by shavitush** [comment]
@Lime18 what map is this?

**Comment by Lime18** [comment]
In the screenshot http://steamcommunity.com/sharedfiles/filedetails/?id=899766046
Errors http://steamcommunity.com/sharedfiles/filedetails/?id=542677765

**Comment by shavitush** [comment]
What if you change `shavit_misc_createspawnpoints` to a low number, like 3? Does the issue solve?

**Comment by Lime18** [comment]
It did not work, shavit_misc_createspawnpoints 3 and shavit_misc_createspawnpoints 0
https://prnt.sc/hcef21

**Comment by shavitush** [comment]
Will investigate soon.

**Comment by SlidyBat** [comment]
I think the block disappearing thing is an mpbh issue, it happens on many other timers as well. I've edited it and haven't experienced the issue since but I'm not sure if it has actually fixed it. Edited version can be found here: https://forums.alliedmods.net/showthread.php?p=2561012#post2561012 

Can you try it out to see if that fixes the issue?

**Comment by Lime18** [comment]
@SlidyBat I will check it!

**Comment by Lime18** [comment]
@SlidyBat it seems like it solved the problem with the blocks, thanks.

But there's still a problem with the triggers ...
Teleport trigger 'teleport stage 4' cannot find destination named 'stage 4'!

**Comment by Lime18** [comment]
@shavitush 
map http://steamcommunity.com/sharedfiles/filedetails/?id=1195619375

https://prnt.sc/hegun1

Teleport trigger '' cannot find destination named 's2'!
Teleport trigger '' cannot find destination named 's2'!
Teleport trigger '' cannot find destination named 's2'!
Teleport trigger '' cannot find destination named 's2'!
Teleport trigger '' cannot find destination named 's2'!

[SM] Listing 10 plugins:
  01 "[shavit] Replay Bot" (1.5b) by shavit
  02 "[shavit] Map Zones" (1.5b) by shavit
  03 "[shavit] Player Stats" (1.5b) by shavit
  04 "[shavit] Dynamic Timelimits" (1.5b) by shavit
  05 "[shavit] HUD" (1.5b) by shavit
  06 "[shavit] Core" (1.5b) by shavit
  07 "[shavit] World Records" (1.5b) by shavit
  08 "[shavit] Miscellaneous" (1.5b) by shavit
  09 "[shavit] Sounds" (1.5b) by shavit
  10 "[shavit] Rankings" (1.5b) by shavit

1. remove shavit-misc.smx, restart **map** - not working!
2. remove shavit-misc.smx, restart **server** - working!

**Comment by shavitush** [comment]
I can't access the Steam workshop :/ hit me up with a .bsp (or .bsp.bz2) and I'll check it.

**Comment by Lime18** [comment]
maybe this will somehow help
https://www.youtube.com/watch?v=lglyR_ZQC9k

**Comment by strafe** [comment]
@shavitush [bhop_kyle.bsp.zip](https://github.com/shavitush/bhoptimer/files/1502552/bhop_kyle.bsp.zip)

**Comment by Lime18** [comment]
The problem was replicated on the local server.
https://www.youtube.com/watch?v=QGHCOnuH2ww
https://www.youtube.com/watch?v=GBkhS0lc3BA
